animations/lib/src/basics/08_animated_list.dart

```diff
@@ -0,0 +1,137 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class AnimatedListDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_list';
+
+  _AnimatedListDemoState createState() => _AnimatedListDemoState();
+}
+
+class _AnimatedListDemoState extends State<AnimatedListDemo> {
+  final GlobalKey<AnimatedListState> _listKey = GlobalKey();
+
+  void addUser() {
+    int index = listData.length;
+    listData.add(listData[Random().nextInt(5)]);
+    _listKey.currentState
+        .insertItem(index, duration: Duration(milliseconds: 500));
+  }
+
+  void deleteUser(int index) {
+    var user = listData.removeAt(index);
+    _listKey.currentState.removeItem(
+      index,
+      (context, animation) {
+        return FadeTransition(
+          opacity:
+              CurvedAnimation(parent: animation, curve: Interval(0.5, 1.0)),
+          child: SizeTransition(
+            sizeFactor:
+                CurvedAnimation(parent: animation, curve: Interval(0.0, 1.0)),
+            axisAlignment: 0.0,
+            child: _buildItem(user),
+          ),
+        );
+      },
+      duration: Duration(milliseconds: 600),
+    );
+  }
+
+  Widget _buildItem(UserModel user, [int index]) {
+    return ListTile(
+      key: ValueKey<UserModel>(user),
+      title: Text(user.firstName),
+      subtitle: Text(user.lastName),
+      leading: CircleAvatar(
+        backgroundImage: NetworkImage(user.profileImageUrl),
+      ),
+      onLongPress: index != null ? () => deleteUser(index) : null,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Animated List Demo"),
+        centerTitle: true,
+        leading: IconButton(
+          icon: Icon(Icons.add),
+          onPressed: addUser,
+        ),
+      ),
+      body: SafeArea(
+        child: AnimatedList(
+          key: _listKey,
+          initialItemCount: listData.length,
+          itemBuilder: (context, index, animation) {
+            return FadeTransition(
+              opacity: animation,
+              child: _buildItem(listData[index], index),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class UserModel {
+  UserModel({this.firstName, this.lastName, this.profileImageUrl});
+
+  String firstName;
+  String lastName;
+  String profileImageUrl;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is UserModel &&
+          runtimeType == other.runtimeType &&
+          firstName == other.firstName &&
+          lastName == other.lastName &&
+          profileImageUrl == other.profileImageUrl;
+
+  @override
+  int get hashCode =>
+      firstName.hashCode ^ lastName.hashCode ^ profileImageUrl.hashCode;
+}
+
+List<UserModel> listData = [
+  UserModel(
+    firstName: "Govind",
+    lastName: "Dixit",
+    profileImageUrl:
+        "https://ca.slack-edge.com/TADUGCD9D-UC5F6HJ6T-gfbe5883d03f-1024",
```

Comment:
This image URL adds a dependency for the sample. If that URL were to ever stop working, it would break the app.

Rather than using an image for the avatar, try using a material Icon instead:

```
CircleAvatar(
   child: Icon(Icons.person),
),
```

The icons are guaranteed to be available, and can be loaded without any network calls. This would also allow you to simplify UserModel from three fields to two.